microk8s-resources/actions/common/utils.sh

```diff
@@ -117,3 +117,11 @@ produce_server_cert() {
     openssl req -new -key ${SNAP_DATA}/certs/server.key -out ${SNAP_DATA}/certs/server.csr -config ${SNAP_DATA}/certs/csr.conf
     openssl x509 -req -in ${SNAP_DATA}/certs/server.csr -CA ${SNAP_DATA}/certs/ca.crt -CAkey ${SNAP_DATA}/certs/ca.key -CAcreateserial -out ${SNAP_DATA}/certs/server.crt -days 100000 -extensions v3_ext -extfile ${SNAP_DATA}/certs/csr.conf
 }
+
+maybe_cache_juju_operator_images() {
+    if [ ! -f /snap/bin/juju.cache-images ]; then
+        echo "No supported version of Juju installed."
+        exit 0
+    fi
+    /snap/bin/juju.cache-images
```

Comment:
What is the cost of this? How much data are we downloading at this point?

Reply:
The entire MicroK8s is 200MB. We would like to keep this number low as it dominates the installation time.

Reply:
It's about 200MB for the 2 images. I can understand the apprehension adding to the install size. Let me find some shareable details around why we want to do this :-)

Reply:
This doubles the deployment time. Obviously not good.

Why do we flip the switch of caching during installation and don't we flip the switch with a `microk8s.enable juju-integration`. This would be the same as with any other addon, I am also pretty sure we already expect a few addons to be available (eg dns storage) so it is not surprising for the user.

Reply:
There is also the option of an auto connect interface. I am afraid this approach of inferring the existence of juju through the /snap/bin may not work when we move to strict confinement.

Reply:
Could this `juju.cache-images` be run on the background?